docs/dev/sql-nested-function-where-clause.md

```diff
@@ -0,0 +1,397 @@
+## Description
+
+The `nested` function when used in the `WHERE` clause of an SQL statement filters documents of `nested` object type based on field properties. Two syntax options are available to users see section [1.1 Syntax](#11-syntax), both syntax options achieve the goal of filtering with a `nested` field, an `Operator` and a `Literal` value. Both syntax options produce different DSL to push to OpenSearch based on the `AND/OR/NOT` boolean queries residing inside a `nested` query or vice versa. See section [4?]() for examples depicting this alternate DSL creation. Testing large groups of push down execution times for identical queries using the different syntax options showed a nominal difference of approximately 3% for Syntax option 2.
+
+## Table of Contents
+1. [Overview](#1-overview)
+2. [Syntax](#11-syntax)
+3. [Changes To Core](#12-changes-to-core)
+4. [Example Data And Usage](#13-example-data-and-usage)
+5. [Example Queries](#14-example-queries)
+6. [Architecture Diagrams](#2-architecture-diagrams)
+7. [Sequence Diagram for filter Push Down](#21-sequence-diagram-for-filter-push-down)
+8. [Syntax Option 1 Object Tree to DSL](#22-syntax-option-1-object-tree-to-dsl)
+9. [Syntax Option 2 Object Tree to DSL](#23-syntax-option-2-object-tree-to-dsl)
+10. [Additional Info](#additional-info)
+11. [Release Schedule](#release-schedule)
+
+## 1 Overview
+### 1.1 Syntax
+
+The nested function has two syntax options when used in the `WHERE` clause of an SQL statement. Both syntax options a
```

Comment:
Was this sentence supposed to end with `Both syntax options a` or was the sentence cut off?

Reply:
Fixed in [fafca288aad64c8c4be3ff8bce2e49668bc7a4d2](https://github.com/Bit-Quill/opensearch-project-sql/pull/265/commits/fafca288aad64c8c4be3ff8bce2e49668bc7a4d2).